chainer/variable.py

```diff
@@ -545,12 +546,12 @@ def _init_unchecked(data=None, name=None, grad=None, requires_grad=True,
 
         # Create a Variable without invoking __init__
         var = Variable.__new__(Variable)
-        var._init_impl(data, name, grad, requires_grad, is_chainerx_array,
+        var._init_impl(None, data, name, grad, requires_grad, is_chainerx_array,
                        node)
         return var
 
-    def _init_impl(self, data, name, grad, requires_grad, is_chainerx_array,
-                   node):
+    def _init_impl(self, device, data, name, grad, requires_grad,
+                   is_chainerx_array, node):
```

Comment:
I think this interface makes it look like any device can be passed. How about renaming it to `chainerx_device`, placing it not as the first argument but later, and then asserting the equality of it with the device of `data`?

Reply:
I see. In that case I'd prefer to assert for all "xp" cases, maybe also with a comment that it's not tested for non-ChainerX `device`-`data` pairs.

Reply:
>assert for all "xp" cases
What does this mean?

Reply:
> I can't see any reason to **restrict it to ChainerX**....

I mean all device types, including `ch.backend.ChainerxDevice`.

Reply:
I mean, what does "asserting for xp" mean?

Reply:
Ah, I meant to assert that the `device` is the correct device associated with `data`.

Reply:
That would be costly. `_init_unchecked` exists for speed.

Reply:
Hm, yes that is true. How about leaving a comment describing the expectations, including the fact that it's there for performance reasons? It would help the next developer reading this code.

Reply:
Added comment.